pyvista/core/dataobject.py

```diff
@@ -189,39 +189,172 @@ def copy(self, deep=True):
         Parameters
         ----------
         deep : bool, optional
-            When True makes a full copy of the object.
+            When ``True`` makes a full copy of the object.  When
+            ``False``, performs a shallow copy where the points, cell,
+            and data arrays are pointers to the original object.
 
         Returns
         -------
-        newobject : same as input
-           Deep or shallow copy of the input.
+        :class:`pyvista.DataSet`
```

Comment:
Is this OK when we're returning a subclass? It's technically correct that the returned object is a `DataSet`, but perhaps we should keep this explicit. So maybe adding it to the description, that the type is that of the input?

Reply:
Done in code suggestion.